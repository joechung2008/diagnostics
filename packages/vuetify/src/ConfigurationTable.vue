<script lang="ts" setup>
import { computed } from 'vue'
import type { ConfigurationProps } from './ConfigurationTable.types'

const props = defineProps<ConfigurationProps>()

const headers = [
  { title: 'Key', value: 'key' },
  { title: 'Value', value: 'value' },
]

const items = computed(() => Object.entries(props.config).map(([key, value]) => ({ key, value })))
</script>

<template>
  <h2 class="text-h4">Configuration</h2>
  <v-data-table :headers="headers" :items="items" hide-default-footer />
</template>
