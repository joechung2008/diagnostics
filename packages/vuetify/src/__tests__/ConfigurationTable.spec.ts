// ConfigurationTable.spec.ts
import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import ConfigurationTable from '../ConfigurationTable.vue'

describe('ConfigurationTable', () => {
  it('renders configuration items', () => {
    const config = {
      apiUrl: 'https://example.com',
      timeout: '5000',
    }
    const wrapper = mount(ConfigurationTable, {
      props: { config },
    })
    expect(wrapper.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
