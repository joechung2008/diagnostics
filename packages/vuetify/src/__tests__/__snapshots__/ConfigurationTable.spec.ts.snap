// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConfigurationTable > renders configuration items 1`] = `
"<h2 class="text-h4">Configuration</h2>
<div class="v-table v-table--has-top v-table--has-bottom v-theme--light v-table--density-default v-data-table">
  <!---->
  <div class="v-table__wrapper">
    <table>
      <!---->
      <thead>
        <tr>
          <th class="v-data-table__td v-data-table-column--align-start v-data-table__th" colspan="1" rowspan="1">
            <div class="v-data-table-header__content"><span>Key</span>
              <!---->
              <!---->
            </div>
          </th>
          <th class="v-data-table__td v-data-table-column--align-start v-data-table__th" colspan="1" rowspan="1">
            <div class="v-data-table-header__content"><span>Value</span>
              <!---->
              <!---->
            </div>
          </th>
        </tr>
        <!---->
      </thead>
      <!---->
      <tbody>
        <!---->
        <tr class="v-data-table__tr">
          <td class="v-data-table__td v-data-table-column--align-start">apiUrl</td>
          <td class="v-data-table__td v-data-table-column--align-start">https://example.com</td>
        </tr>
        <!---->
        <tr class="v-data-table__tr">
          <td class="v-data-table__td v-data-table-column--align-start">timeout</td>
          <td class="v-data-table__td v-data-table-column--align-start">5000</td>
        </tr>
        <!---->
        <!---->
      </tbody>
      <!---->
      <!---->
    </table>
  </div>
  <!---->
</div>"
`;
